static/src/javascripts/projects/facia/modules/onwards/weather.js

```diff
@@ -0,0 +1,212 @@
+import mediator from 'lib/mediator';
+import userPrefs from 'common/modules/user-prefs';
+import { SearchTool } from 'facia/modules/onwards/search-tool';
+import type { CityPreference } from 'facia/modules/onwards/search-tool';
+
+let $holder = null;
+let searchTool = null;
+let eventsBound = false;
+const prefName = 'weather-location';
+
+type Location = {
+    id: string,
+    city: string,
+};
+
+type WeatherResponse = {
+    html: string,
+};
+
+const isNetworkFront = () =>
+    ['uk', 'us', 'au', 'international'].includes(config.page.pageId);
+
+export const Weather = {
+    init() {
+        if (!config.switches || !config.switches.weather || !isNetworkFront()) {
+            return false;
+        }
+
+        this.getDefaultLocation();
+    },
+
+    isNetworkFront() {
+        return ['uk', 'us', 'au', 'international'].includes(config.page.pageId);
+    },
+
+    /**
+     * Check if user has data in local storage.
+     * If yes return data from local storage else return default location data.
+     *
+     * @returns {object} geolocation - lat and long
+     */
+    getUserLocation() {
+        const prefs = userPrefs.get(prefName);
+
+        if (prefs && prefs.id) {
+            return prefs;
+        }
+    },
+
+    getWeatherData(url: string) {
+        return fetchJson(url, {
+            mode: 'cors',
+        });
+    },
+
+    /**
+     * Save user location into localStorage
+     */
+    saveUserLocation(location: Location) {
+        userPrefs.set(prefName, {
+            id: location.id,
+            city: location.city,
+        });
+    },
+
+    getDefaultLocation() {
+        const location = this.getUserLocation();
+
+        if (location) {
+            this.fetchWeatherData(location);
+        } else {
+            return this.getWeatherData(`${config.page.weatherapiurl}.json`)
+                .then(response => {
+                    this.fetchWeatherData(response);
+                })
+                .catch(err => {
+                    reportError(err, {
+                        feature: 'weather',
+                    });
+                });
+        }
+    },
+
+    fetchWeatherData(location: Location) {
+        return this.getWeatherData(
+            `${config.page.weatherapiurl}/${location.id}.json?_edition=${config.page.edition.toLowerCase()}`
+        )
+            .then(response => {
+                this.render(response, location.city);
+                this.fetchForecastData(location);
+            })
+            .catch(err => {
+                reportError(err, {
+                    feature: 'weather',
+                });
+            });
+    },
+
+    clearLocation() {
+        userPrefs.remove(prefName);
+        if (searchTool !== null) {
+            searchTool.setInputValue();
+        }
+    },
+
+    fetchForecastData(location: Location) {
+        return this.getWeatherData(
+            `${config.page.forecastsapiurl}/${location.id}.json?_edition=${config.page.edition.toLowerCase()}`
```

Comment:
Could you move this out of the string and save it as a variable, to make it more readable?